test/unit/math/mix/fun/quad_form_sym_test.cpp

```diff
@@ -64,3 +64,40 @@ TEST(MathMixMatFun, quadFormSym) {
   v << 100, 10, 0, 1, -3, -3, 5, 2;
   stan::test::expect_ad(g, u, v);
 }
+
+TEST(MathMixMatFun, quad_form_sym_2095) {
+  Eigen::Matrix<stan::math::var, -1, -1> av(2, 2);
+  Eigen::Matrix<stan::math::var, -1, -1> bv(2, 2);
+
+  av << 1.25882993696386514, -0.03325949401909023, -0.03325949605426004,
+      1.85523447220884385;
+
+  bv << 2.55474619740069508, 0.66362927717720988, -1.92917223922387349,
+      2.01853255256721731;
+
+  Eigen::Matrix<stan::math::var, -1, -1> cv = stan::math::quad_form_sym(av, bv);
+  EXPECT_FLOAT_EQ(0, cv(1, 0).val() - cv(0, 1).val());
+
+  stan::math::recover_memory();
+
+  //--------------------
+
+  Eigen::MatrixXd ad(2, 2);
+  Eigen::MatrixXd bd(2, 2);
+
+  ad << 1.25882993696386514, -0.03325949401909023, -0.03325949605426004,
```

Comment:
This matrix isn't symmetric.

Reply:
Line 99 makes it symmetric.

Reply:
This was an actual matrix in the wild that triggered the bug. I noticed that non-symmetry too, but it’s within tolerance, so it’s ok. This matrix gets used directly in `prim` and gets checked for symmetry.

Reply:
Oh I didn't realize symmetric was only symmetric to an absolute tolerance of 1e-8.

I wonder if this is the source of the sporadic symmetric errors that show up in models. From here: https://discourse.mc-stan.org/t/is-it-possible-to-disable-symmetry-checks/6544

Stuff like:
```
Chain 13: Exception: Exception: quad_form_sym: A is not symmetric. A[56,74] = -4.43735e+07, but A[74,56] = -4.43735e+07  (in 'modelb0b43b4c742c_model6_marginalize_interp' at line 178)
```

There are some other symmetry checks in that thread that I am surprised are failing though, so maybe it's not the whole story.

I suspect if we switched to using the tolerance stuff from the test framework this would get rid of most of these sporadic problems: https://github.com/stan-dev/math/blob/develop/test/unit/math/expect_near_rel.hpp .

Reply:
> Oh I didn't realize symmetric was only symmetric to an absolute tolerance of 1e-8.

Yup. We can't really do `==` on floating point numbers.

> There are some other symmetry checks in that thread that I am surprised are failing though, so maybe it's not the whole story.

I'd guess there's more to it than just the symmetry checks.

> I suspect if we switched to using the tolerance stuff from the test framework this would get rid of most of these sporadic problems: https://github.com/stan-dev/math/blob/develop/test/unit/math/expect_near_rel.hpp .

+1! That'd be great. It really should be a relative tolerance instead of an absolute tolerance. But this brings up a practical issue. If functions assume symmetry, but use both halves of the matrix, then the errors could accumulate. There are different approaches we can take to clean this up...

Reply:
> There are different approaches we can take to clean this up

Expand on this.

I want to open an issue for making check_symmetric use `expect_near_real` but before I do that I wanna know about these other things.